mscore/musescoredialogs.cpp

```diff
@@ -137,7 +137,7 @@ void AboutBoxDialog::copyRevisionToClipboard()
       // endianness?
       sysinfo += ", MuseScore version (";
       sysinfo += QSysInfo::WordSize==32 ? "32" : "64";
-      auto msVersion = QString(VERSION) + QString(".") + QString(BUILD_NUMBER);
+      auto msVersion = QString(VERSION) + (QString(BUILD_NUMBER).size() ? QString(".") + QString(BUILD_NUMBER) : QString(""));
```

Comment:
1. Why don't you use QString::isEmpty()?
    It is more readable and doesn't require implicit conversions from int type
2. I can see at least 2 malloc of BUILD_NUMBER string here. It would be better to avoid it:
    - You'll just create a local variable (recommended)
    - You'll implement this check in 'preprocessor' way

P.S.
In general, most of the file should be processed using QStringBuilder, but we will return to this later.

Reply:
1. I can certainly use `QString::isEmpty()`.
2. No `malloc()` in sight, this is C++, so it'd be `new` ;-)

    * Not really needed I think. There's even one less now.
    * I can't see how to do with with preprocessos, BUILD_NUMBER is #define'd as "", no (reasonably straight forward) way to check for an empty string using preprocessor directives.

Reply:
> No malloc() in sight, this is C++, so it'd be new ;-)
This is Qt and internaly it'd copy chars and reallocate memmory blocks through the malloc for every concat operation.

Reply:
I'm pretty sure Qt uses C++ internally, but yes I get your point that some form of memory allocation happens here and without real need.

Reply:
> I'm pretty sure Qt uses C++ internally

Oh, finally, I realized what was the misunderstanding when you talked about C and C++.
In C++ we also have std::malloc() and it's used widely in low-level programming (SDK, firmwares, etc.)

Reply:
Low level C++ is called C ;-)